pinot-controller/src/main/java/org/apache/pinot/controller/util/ServerSegmentMetadataReader.java

```diff
@@ -140,16 +143,17 @@ public TableMetadataInfo getAggregatedTableMetadataFromServer(String tableNameWi
       return v;
     });
 
-    // Since table segments may have multiple replicas, divide diskSizeInBytes, numRows and numSegments by numReplica
-    // to avoid double counting, for columnAvgLengthMap, columnAvgCardinalityMap and maxNumMultiValuesMap, dividing by
-    // numReplica is not needed since totalNumSegments already contains replicas.
+    // Since table segments may have multiple replicas, divide diskSizeInBytes, numRows, numSegments and primary key
+    // count by numReplica to avoid double counting, for columnAvgLengthMap, columnAvgCardinalityMap and
+    // maxNumMultiValuesMap, dividing by numReplica is not needed since totalNumSegments already contains replicas.
     totalDiskSizeInBytes /= numReplica;
     totalNumSegments /= numReplica;
     totalNumRows /= numReplica;
+    upsertPartitionToPrimaryKeyCountMap.replaceAll((k, v) -> v / numReplica);
```

Comment:
I think (and the preceding calculations for totalDiskSizeInBytes, etc.) this assumes that each replica will have the exact same number of PKs as the other replicas, but what happens if that is not the case (ie, one replica is inconsistent and has a different number of PKs)? That discrepancy will get lost here but could be very valuable to have.

Is it possible for `upsertPartitionToPrimaryKeyCountMap` to store PK count for each Replica of each partition/segment?

Reply:
@klsince @ege-st I agree to above point. Adding replica-level information can help improve the usability of the API.

I can return partition to server to PK count Map 

{
"0": {
            "Server1":50,
            "Server2:28,
        },
},

Reply:
I think this would be great.  Thanks.